core/scripts/common/helpers.go

```diff
@@ -0,0 +1,29 @@
+package common
+
+import (
+	"flag"
+	"fmt"
+)
+
+func PanicErr(err error) {
+	if err != nil {
+		panic(err)
+	}
+}
+
+func ParseArgs(flagSet *flag.FlagSet, args []string, requiredArgs ...string) {
+	PanicErr(flagSet.Parse(args))
+	seen := map[string]bool{}
+	argValues := map[string]string{}
+	flagSet.Visit(func(f *flag.Flag) {
+		seen[f.Name] = true
+		argValues[f.Name] = f.Value.String()
+	})
+	for _, req := range requiredArgs {
+		if !seen[req] {
+			panic(fmt.Errorf("missing required -%s argument/flag", req))
+		} else if req == "sub-id" && argValues[req] == "0" {
+			panic(fmt.Errorf("sub-id must be nonzero"))
```

Comment:
Why does sub-id need special handling? Are you worried about people explicitly setting sub-id to zero?

Reply:
Yes and also by default sub-is is set to zero. `subID := request.Uint64("sub-id", 0, "subscription ID")` Adding validation to verify that its non-zero.

Reply:
fixed.